qadi_multi_perspective.py

```
@@ -0,0 +1,299 @@
+    # Fallback if package is not installed
+    sys.path.insert(0, str(Path(__file__).parent / "src"))
+    from mad_spark_alt.core import setup_llm_providers
+    from mad_spark_alt.core.intent_detector import IntentDetector, QuestionIntent
+    from mad_spark_alt.core.multi_perspective_orchestrator import (
+        MultiPerspectiveQADIOrchestrator,
+        PerspectiveResult,
+    )
+    from mad_spark_alt.core.terminal_renderer import render_markdown
+
+
+def display_intent_detection(intent_result: 'IntentResult', perspectives: List[QuestionIntent]) -> None:
```

Ruff (0.12.2)

Undefined name `IntentResult`

(F821)